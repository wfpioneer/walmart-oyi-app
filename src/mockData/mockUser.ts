import User from '../models/User';
import { mockConfig } from './mockConfig';

const mockUser: User = {
  userId: 'testUser123',
  additional: {
    clockCheckResult: 'yes',
    displayName: 'test user',
    loginId: 'testUser123',
    mailId: 'testUser123@homeoffice.wal-mart.com'
  },
  configs: mockConfig,
  countryCode: 'US',
  domain: 'Home Office',
  features: [],
  siteId: 6233,
  token: 'fakeToken'
};

export default mockUser;
