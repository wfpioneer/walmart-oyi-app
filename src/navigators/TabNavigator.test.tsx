import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockUser from '../mockData/mockUser';
import {
  TabNavigatorStack, tabBarOptions
} from './TabNavigator';

jest.mock('../locales', () => ({
  ...jest.requireActual('../locales'),
  strings: (key: string) => key
}));

jest.mock('@react-navigation/bottom-tabs', () => ({
  createBottomTabNavigator: jest.fn(() => ({
    Navigator: jest.fn(),
    Screen: jest.fn()
  }))
}));

describe('Tab Navigator', () => {
  it('Render ReviewItemsNavigator', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <TabNavigatorStack
        user={mockUser}
        selectedAmount={0}
        dispatch={jest.fn()}
        palletWorklists={false}
        auditWorklists={false}
        isBottomTabEnabled={false}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('Render ReviewItemsNavigator with user features', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <TabNavigatorStack
        user={mockUser}
        selectedAmount={0}
        dispatch={jest.fn()}
        palletWorklists={true}
        auditWorklists={true}
        isBottomTabEnabled={false}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders TabNavigatorStack with isBottomTabEnabled as true', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <TabNavigatorStack
        user={mockUser}
        selectedAmount={0}
        dispatch={jest.fn()}
        palletWorklists={true}
        auditWorklists={true}
        isBottomTabEnabled={true}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Should return the expected options from tabBarOptions', () => {
    const props = {
      selectedAmount: 0,
      user: mockUser,
      dispatch: jest.fn(),
      palletWorklists: true,
      auditWorklists: true,
      isBottomTabEnabled: true
    };
    const route = {
      key: 'Home',
      name: 'HOME.HOME'
    };
    const result = tabBarOptions(props, route);

    expect(result).toEqual({
      tabBarIcon: expect.any(Function),
      tabBarStyle: { display: 'flex' },
      tabBarActiveTintColor: expect.any(String),
      tabBarInactiveTintColor: expect.any(String),
      headerShown: false
    });
  });
});
