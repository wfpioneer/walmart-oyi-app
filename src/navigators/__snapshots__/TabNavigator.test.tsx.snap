// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Navigator Render ReviewItemsNavigator 1`] = `
<mockConstructor
  screenOptions={[Function]}
>
  <mockConstructor
    component={
      {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    name="HOME.HOME"
  />
  <mockConstructor
    component={[Function]}
    name="WORKLIST.WORKLIST"
  />
</mockConstructor>
`;

exports[`Tab Navigator Render ReviewItemsNavigator with user features 1`] = `
<mockConstructor
  screenOptions={[Function]}
>
  <mockConstructor
    component={
      {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    name="HOME.HOME"
  />
  <mockConstructor
    component={[Function]}
    name="WORKLIST.WORKLIST"
  />
</mockConstructor>
`;

exports[`Tab Navigator Renders TabNavigatorStack with isBottomTabEnabled as true 1`] = `
<mockConstructor
  screenOptions={[Function]}
>
  <mockConstructor
    component={
      {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "type": [Function],
      }
    }
    name="HOME.HOME"
  />
  <mockConstructor
    component={[Function]}
    name="WORKLIST.WORKLIST"
  />
</mockConstructor>
`;
